modules/oe_theme_helper/tests/src/Kernel/Plugin/Field/FieldFormatter/MediaThumbnailUrlFormatterTestBase.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme_helper\Kernel\Plugin\Field\FieldFormatter;
+
+use Drupal\media\Entity\Media;
+use Drupal\Tests\media\Traits\MediaTypeCreationTrait;
+use Drupal\Tests\oe_theme\Kernel\AbstractKernelTestBase;
+
+/**
+ * Base class for formatters rendering media thumbnail URLs.
+ */
+class MediaThumbnailUrlFormatterTestBase extends AbstractKernelTestBase {
+
+  use MediaTypeCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'field',
+    'entity_test',
+    'media',
+    'image',
+    'file',
+    'entity_reference',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->installEntitySchema('entity_test');
+    $this->installEntitySchema('media');
+    $this->installEntitySchema('file');
+    $this->installSchema('file', 'file_usage');
+
+    $this->installConfig([
+      'file',
+      'field',
+      'entity_reference',
+      'media',
+    ]);
+  }
+
+  /**
+   * Create a media of type "image".
+   *
+   * @param string $filepath
+   *   Full path to image file.
+   *
+   * @return \Drupal\media\Entity\Media
+   *   Media object.
+   */
+  protected function createMediaImage(string $filepath): Media {
```

Comment:
We need methods like: 
* createMediaImage($name, $filepath) (https://github.com/openeuropa/oe_theme/blob/EPIC-project/tests/Functional/ContentProjectRenderTest.php#L539),
* createMediaDocument($name, $filepath) https://github.com/openeuropa/oe_theme/blob/EPIC-project/tests/Functional/ContentProjectRenderTest.php#L569,
* createOrganisationEntity($name, $bundle)  (https://github.com/openeuropa/oe_theme/blob/EPIC-project/tests/Functional/ContentProjectRenderTest.php#L331)
almost in all content type tests.

Also asserts of field formatters are re-used a lot for 
* Media Document Default view mode (https://github.com/openeuropa/oe_theme/pull/529/files#diff-07c9677f4102faefc99f738124eaa57aR85)
* Featured Media field (https://github.com/openeuropa/oe_theme/blob/EPIC-project/tests/Functional/ContentProjectRenderTest.php#L597)

I think we need some one place where all these methods will be stored. It can be base class in oe_theme/src/tests/.../ (but if we have Kernel and Functional tests both not sure it can be implemented) or trait classes in oe_theme/src/tests/ that can be included to the tests or another option.

I see that you have started with assertions based on BasePatternAssert in this PR. My initiative was https://github.com/openeuropa/oe_theme/pull/529/files#diff-2b3a320550cf248ff78b8e0da8355d64R15 that can be re-used for Event/Project/Tender content types but looks like such approach isn't correct as we will have Kernel and Functional tests both. Anyway we also need createEntity() methods to reduce code duplication and simplify developer's work

Reply:
I think the best is to organise all of those into traits, stored in in the module's codebase, so that they can be used on all types of tests, kernel, functional, behat contexts, etc. let's do this gradually, after release we will focus on a full test rewrite on oe_theme and oe_content, so that we consolidate by using the same techniques everywhere.